bdk_chain/tests/common/mod.rs

```diff
@@ -48,3 +50,54 @@ macro_rules! changeset {
         }
     }};
 }
+
+#[allow(unused)]
+pub fn must_insert_tx<P: bdk_chain::sparse_chain::ChainPosition>(
+    cg: &mut bdk_chain::chain_graph::ChainGraph<P>,
+    tx: bitcoin::Transaction,
+    pos: P,
+) {
+    let changeset = cg.insert_tx_changeset(tx, pos).expect("must insert tx");
+    cg.apply_changeset(changeset).expect("must insert tx")
+}
+
+#[allow(unused)]
+pub fn must_insert_checkpoint<P: bdk_chain::sparse_chain::ChainPosition>(
+    cg: &mut bdk_chain::chain_graph::ChainGraph<P>,
+    block_id: bdk_chain::BlockId,
+) {
+    let changeset = cg
+        .insert_checkpoint_changeset(block_id)
+        .expect("must insert checkpoint");
+    cg.apply_changeset(changeset)
+        .expect("must insert checkpoint")
+}
+
+#[allow(unused)]
+pub fn graph_insert_tx(graph: &mut bdk_chain::tx_graph::TxGraph, tx: bitcoin::Transaction) {
+    let additions = graph.insert_tx_changeset(tx);
+    graph.apply_additions(additions);
+}
+
+#[allow(unused)]
+pub fn chain_insert_tx<P: bdk_chain::sparse_chain::ChainPosition>(
+    chain: &mut bdk_chain::sparse_chain::SparseChain<P>,
+    txid: Txid,
+    pos: P,
+) {
+    let changeset = chain
+        .insert_tx_changeset(txid, pos)
+        .expect("must insert tx");
+    chain.apply_changeset(changeset);
+}
+
+#[allow(unused)]
+pub fn chain_insert_checkpoint<P: bdk_chain::sparse_chain::ChainPosition>(
+    chain: &mut bdk_chain::sparse_chain::SparseChain<P>,
+    block_id: bdk_chain::BlockId,
+) {
+    let changeset = chain
+        .insert_checkpoint_changeset(block_id)
+        .expect("must insert tx");
+    chain.apply_changeset(changeset);
+}
```

Comment:
These functions seem to exist because the API currently has a lot of friction because of the two-step changeset. If it's worth getting rid of this friction while testing it's probably worth getting rid of it for users too.